File: test-projects/test-java-project/src/main/java/com/github/vilmosnagy/elq/testproject/entities/Album.java
package com.github.vilmosnagy.elq.testproject.entities;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.util.Date;

/**
 * @author Vilmos Nagy {@literal <vilmos.nagy@outlook.com>}
 */
@Entity
@Data
@EqualsAndHashCode(exclude = "artist")
@Table(name = "Album")
public class Album {

    @Id
    @Column(name="AlbumId")
    private int id;

    @Version
    @Column(name = "version")
    private Date version;

    @Column(name = "title")
    private String title;

    @ManyToOne
    @JoinColumn(name = "artistId", referencedColumnName = "artistId")
    private Artist artist;

}

File: test-projects/test-kotlin-project/src/test/kotlin/com/github/vilmosnagy/elq/testproject/entities/Album.kt
package com.github.vilmosnagy.elq.testproject.entities

import java.util.*
import javax.persistence.*

/**
 * @author Vilmos Nagy {@literal <vilmos.nagy@outlook.com>}
 */
@Entity
@Table(name = "Album")
data class Album private constructor(
        @Id @Column(name="AlbumId")         override var id: Int,
        @Version @Column(name = "version")  override var version: Date,
        @Column(name = "title")             var title: String
): IBaseEntity {

    constructor(id: Int, version: Date, title: String, artist: Artist): this(id, version, title) {
        this.artist = artist
    }

    @ManyToOne
    @JoinColumn(name = "artistId", referencedColumnName = "artistId")
    lateinit var artist: Artist
}